components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -223,14 +235,25 @@ abstract class AbstractFetchDownloadService : Service() {
         val foregroundServiceId = downloadJobs[download.id]?.foregroundServiceId ?: Random.nextInt()
 
         // Create a new job and add it, with its downloadState to the map
-        downloadJobs[download.id] = DownloadJobState(
+        val downloadJobState = DownloadJobState(
```

Comment:
Since this class is getting pretty big, we should probably move `DownloadJobState` to its own file

Reply:
I think we can do this in follow up, as it will cause the diff more difficult to be reviewed. What do you think?

Reply:
Sounds good to me!